process/block/preprocess/gasComputation.go

```diff
@@ -170,6 +184,38 @@ func (gc *gasComputation) ComputeGasConsumedByTx(
 		return 0, 0, process.ErrNilTransaction
 	}
 
+	if !gc.flagGasComputeV2.IsSet() {
+		return gc.computeGasConsumedByTxV1(txSenderShardId, txReceiverShardId, txHandler)
+	}
+
+	moveBalanceConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
+
+	_, txCrossShard := gc.txTypeHandler.ComputeTransactionType(txHandler)
+	isSCCall := txCrossShard == process.SCDeployment ||
+		txCrossShard == process.SCInvoking ||
+		txCrossShard == process.BuiltInFunctionCall
+	if isSCCall {
+		isCrossShardSCCall := txSenderShardId != txReceiverShardId &&
+			moveBalanceConsumption < txHandler.GetGasLimit() &&
+			txCrossShard != process.BuiltInFunctionCall
+		if isCrossShardSCCall {
+			gasConsumedByTxInSenderShard := moveBalanceConsumption
+			gasConsumedByTxInReceiverShard := txHandler.GetGasLimit() - moveBalanceConsumption
+
+			return gasConsumedByTxInSenderShard, gasConsumedByTxInReceiverShard, nil
+		}
+
+		return txHandler.GetGasLimit(), txHandler.GetGasLimit(), nil
+	}
+
+	return moveBalanceConsumption, moveBalanceConsumption, nil
+}
+
+func (gc *gasComputation) computeGasConsumedByTxV1(
```

Comment:
This function remained untested since all tests are now done on the newer version. Please add unit tests for this part. Might reuse the same unit tests but with a non-zero value for epoch activation

Reply:
done